### PR TITLE
[FIF-314] Public Demo Site - Reset Data.

### PR DESCRIPTION
**Pull request by DavidJGapCR**
### About the solution.
I have added a new migration to the database project. This new script is not part of the default collection. We are using the scoping feature of the db-migrate tool to conditionally execute it. 

A new flag value has been added to the **example.configuration.json** file with the name **loadSampleData**. 

### Testing.
To test it, just execute an installation after changing the **loadSampleData** value in your local **configuration.json** file.

Based on this configuration value, the database should or shouldn't have the sample data.

### CODE REVIEW

File: EdFi.Buzz.Database/migrations/sample-data/20201022195215-adds-sample-data.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+var dbm;
+var type;
+var seed;
+var fs = require('fs');
+var path = require('path');
+var Promise;
```
Comment (kurtjohnson-HS) [inline EdFi.Buzz.Database/migrations/sample-data/20201022195215-adds-sample-data.js:8]: This is surprising to me. Do we need a Promise variable? I would have thought the Promise object was enough.

**Review by kurtjohnson-HS — CHANGES_REQUESTED** [review]
I am still trying to run the functionality, but wanted to request a change, or response to my Promise question in the meantime.

File: EdFi.Buzz.Database/migrations/sample-data/20201022195215-adds-sample-data.js (hunk 1 above)
Comment (DavidJGapCR) [reply EdFi.Buzz.Database/migrations/sample-data/20201022195215-adds-sample-data.js:8]: This is the file generated by db-migrate. Once generated I didn't change anything. So, I guess the answer is Yes. hehe....

File: EdFi.Buzz.Database/eng/Windows/install.ps1 [hunk 2]
```diff
@@ -69,6 +72,22 @@ function Install-Migrations {
     }
 }
 
+function Install-SampleData {
+    if ($LoadSampleData) {
+        try {
+            Push-Location -Path $distFolder
+            Write-Host "Installing sample data" -ForegroundColor Magenta
+            &db-migrate up:sample-data --silent "$DbName" --config ./migrate-database.json | Out-File -FilePath $logFile -Append
```
Comment (kurtjohnson-HS) [inline EdFi.Buzz.Database/eng/Windows/install.ps1:80]: This does not work unless the user has db-migrate installed globall from the command line.

Instead, I would add a yarn task in package.json - `sample-data` - and run it like our migrate function...

```json
"sample-data": "db-migrate up:sample-data --silent --config ./migrate-database.json",
```

Then, line 80 would become...

```PowerShell
&npm run sample-data | Out-File -FilePath $logFile -Append
```

File: EdFi.Buzz.Database/eng/Windows/install.ps1 [hunk 3]
```diff
@@ -69,6 +72,22 @@ function Install-Migrations {
...
+            Write-Host "Sample data installed" -ForegroundColor Magenta
+            Pop-Location
+        }
+        catch {
+            Write-Error "Sample data was not installed"
+            throw
```
Comment (kurtjohnson-HS) [inline EdFi.Buzz.Database/eng/Windows/install.ps1:86]: A couple of things I learned from FIF-299.

Favor `$_` in the catch block for handling exceptions. You can Write-Host with the ScriptStackTrace to produce the error that was being swallowed when I ran.

Also, do use Write-Host over Write-Error. In Octopus Deploy, the ErrorActionPreference is Stop. That means the first Write-Error will cause Octopus to stop execution of the PowerShell script. Then, we can throw `$_` just like the other catch blocks to report the exception up the trace.

**Review by kurtjohnson-HS — CHANGES_REQUESTED** [review]
I wasn't able to run it directly at first because the `Assert-NodeJs` failed, even though I did have Node JS installed. `npm -l` doesn't return anything on my new machine. `npm -l` does return something and passes the test on the Octopus' build machine. `npm` will also return something.

Can also you fix the EdFi.Buzz.Installer `Assert-NodeJs` function? The `npm -l` function I am using is not working to detect node on my new machine. Just checking npm does work.

```PowerShell
    if (Get-Command npm -errorAction SilentlyContinue) {
        $npmVer = npm -l // CHANGE THIS....
        $npmVer = npm // TO THIS...
    }
```
That above change worked for me to get past the Assert-NodeJs check.

**Review by kurtjohnson-HS — APPROVED** [review]

**Pull request merged**